### PR TITLE
Revise platform tags spec, add musllinux

### PR DESCRIPTION
**Pull request by jeanas**
Contains commits:

- Platform tags: Adjust to centralize history

  Move history to a dedicated section (consistent with other specs). In
the main text, do not reference PEPs but just describe the current spec.
Also sketch out a few missing details.

- Remove note about manylinux_x_y being a new standard

  It's been 4 years now.

- Platform tags: Flatten deeply nested headers

- Add musllinux platform tags from PEP 656 (relates to #1093)

----
:books: Documentation preview :books:: https://python-packaging-user-guide--1441.org.readthedocs.build/en/1441/

### CODE REVIEW

**Review by sinoroc — APPROVED** [review]

File: source/specifications/platform-compatibility-tags.rst [hunk 1]
```diff
@@ -89,50 +78,52 @@ revision and compiler flags, etc, but will probably not have a great need
...
-future-proof standard. It defines ``x`` and ``y`` as glibc major an minor
-versions supported (e.g. ``manylinux_2_24`` should work on any distro using
-glibc 2.24+). Previous tags are still supported for backward compatibility.
+* ``manylinux1`` is based on a compatible Linux platform from 2007,
+  and supports ``x86_64`` and ``i686`` architectures.
+* ``manylinux2010`` is based on a platform from 2010 and supports ``x86_64``
+  and ``i686``.
+* ``manylinux2014`` is based on a platform from 2014 and supports
+  ``x86_64``, ``i686``, ``aarch64``, ``armv7l``, ``ppc64``, ``ppc64le``,
+  and ``s390x``.
```
Comment (brettcannon) [inline source/specifications/platform-compatibility-tags.rst:126]: There's actually a mapping of these old manylinux tags back to their equivalent glibc versions at https://github.com/pypa/packaging/blob/769823740a8f5e6b0e37fd05f57f091b56b0ceeb/src/packaging/_manylinux.py#L197C1-L204C2

**Review by brettcannon — APPROVED** [review]

File: source/specifications/platform-compatibility-tags.rst (hunk 1 above)
Comment (jeanas) [reply source/specifications/platform-compatibility-tags.rst]: Oh, thanks, I've made the description more precise with this info.

**Review by sinoroc — APPROVED** [review]

**Comment by webknjaz** [comment]
@jeanas apparently, there's some merge conflict in this branch now.

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
Approving and hoping that the conflict resolution won't reset it.

**Comment by jeanas** [comment]
Fixed the conflict. I think you will have to re-press "Merge".

**Comment by jeanas** [comment]
Thanks everyone for the reviews!

**Pull request merged**